cunumeric/linalg/solve.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021-2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from cunumeric.config import CuNumericOpCode
+
+from legate.core import types as ty
+
+from .exception import LinAlgError
+
+if TYPE_CHECKING:
+    from legate.core.context import Context
+    from legate.core.store import Store
+
+    from ..deferred import DeferredArray
+
+
+def transpose_copy_single(
+    context: Context, input: Store, output: Store
+) -> None:
+    task = context.create_auto_task(CuNumericOpCode.TRANSPOSE_COPY_2D)
+    task.add_output(output)
+    task.add_input(input)
+    # Output has the same shape as input, but is mapped
+    # to a column major instance
+    task.add_scalar_arg(False, ty.int32)
+
+    task.execute()
+
+
+def solve_single(context: Context, a: Store, b: Store) -> None:
+    task = context.create_auto_task(CuNumericOpCode.SOLVE)
+    task.throws_exception(LinAlgError)
+    task.add_input(a)
+    task.add_input(b)
+    task.add_output(a)
+    task.add_output(b)
+    task.execute()
+
+
+def solve(output: DeferredArray, a: DeferredArray, b: DeferredArray) -> None:
+
+    runtime = output.runtime
+    context = output.context
+
+    if runtime.num_procs > 1 or runtime.num_gpus > 1:
```

Comment:
this is bad. a `solve` implementation should do the right thing regardless of the processor count, but this code doesn't do anything if there's more than one processor. you should instead add broadcasting constraints on inputs and outputs to all the tasks in this file to make sure they are never partitioned. then you don't need this if statement.

Reply:
added broadcasting constrains on the input/output arrays and removed this limitation.